File: crates/chidori-js/benchmarks/workloads/string_build.js
// String building (concatenation + number→string coercion).
// Mirrors the `string_build` criterion micro-benchmark, scaled up.
// Kept modest because naive `+=` string building is O(n²) on an engine without
// rope/cord string representation; chidori-js builds the string eagerly.
const N = 30_000;
let s = "";
for (let i = 0; i < N; i++) s += "x" + i;
console.log("RESULT=" + s.length);

File: crates/chidori-js/benchmarks/workloads/string_build.py
# Python twin of string_build.js — string building via += in a loop
# (concatenation + number→string coercion). CPython's in-place str realloc
# optimization makes this its natural fast path, just as ropes would for a JS
# engine — both are fair game, it's the same idiom.
N = 30_000
s = ""
for i in range(N):
    s += "x" + str(i)
print("RESULT=" + str(len(s)))
